source/common/http/http3/conn_pool.h

```diff
@@ -97,6 +98,13 @@ class ActiveClient : public MultiplexedActiveClientBase {
   uint64_t quiche_capacity_ = 100;
 };
 
+class PoolConnectResultCallback {
+public:
+  virtual ~PoolConnectResultCallback() = default;
+  virtual void onConnectSucceeded() PURE;
```

Comment:
Since this class is HTTP/3 specific, I might be inclined to make these method a bit more HTTP/3 specific. In particular, this is about the QUIC/TLS handshake succeeding, right? In QUIC parlance, I believe this is either "Confirmed" or "Finished". So how about:

onHandshakeConfirmed()
onHandshakeFailedWithEarlyData()

(The latter could also be onHandshakeFailed(bool was_early_data_sent) if you prefer)

Reply:
onHandshakeComplete sounds perfect to me.

Reply:
done